build-tools/src/main/java/org/elasticsearch/gradle/plugin/PluginBuildPlugin.java

```diff
@@ -51,6 +58,34 @@ public void apply(final Project project) {
             task.getOutputFile().set(file);
         });
 
-    }
+        project.getTasks().withType(GenerateTestBuildInfoTask.class).named("generateTestBuildInfo").configure(task -> {
+            var jarTask = project.getTasks().withType(Jar.class).named("jar").get();
+            String moduleName = (String) jarTask.getManifest().getAttributes().get("Automatic-Module-Name");
+            if (moduleName == null) {
+                moduleName = jarTask.getArchiveBaseName().getOrNull();
+            }
+            if (moduleName != null) {
+                task.getModuleName().set(moduleName);
+            }
+            var propertiesExtension = project.getExtensions().getByType(PluginPropertiesExtension.class);
+            task.getComponentName().set(providerFactory.provider(propertiesExtension::getName));
+            task.getOutputFile().set(project.getLayout().getBuildDirectory().file("generated-build-info/plugin-test-build-info.json"));
+        });
 
+        project.getTasks().withType(ProcessResources.class).named("processResources").configure(task -> {
+            task.into(
+                (Callable<String>) () -> "META-INF/es-plugins/"
+                    + project.getExtensions().getByType(PluginPropertiesExtension.class).getName()
+                    + "/",
+                copy -> {
+                    copy.from(
+                        (Callable<TaskProvider<GeneratePluginPropertiesTask>>) () -> project.getTasks()
```

Comment:
I actually don't think we need to wrap this one in a `Callable`. A `TaskProvider` is already lazy, so we can just pass that directly.

Reply:
That makes sense! Thanks for all the help with gradle on this PR.